log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -190,9 +190,9 @@ private void bufferDrainLoop()
                 }
                 finally {
                     if (message != null || !dropSnapshot.isEmpty()) {
-                        // Return messages that failed
-                        queueDrainLock.lock();
                         try {
+                            // Return messages that failed
+                            queueDrainLock.lock();
                             // Return a failed message to the queue, or drop it if at capacity.
```

Comment:
Why? It's correct to acquire the lock outside of the try block, which is there to ensure it's unlocked if anything fails:

```java
lock();
try {
   ...
}
finally {
    unlock();
}
```